src/server/controller/health.ts

```diff
@@ -0,0 +1,103 @@
+import { Request, Response } from 'express';
+import { AppDataSource } from '../infrastructure/database';
+import { createNewBlobStorageClient } from '../infrastructure/azure-blob-storage-client';
+import { ExternalServiceError } from '../infrastructure/errors';
+
+interface HealthCheckResult {
+  name: string;
+  label: string;
+  notificationMessage: string | null;
+  shortSummary: string;
+  status: 'ok' | 'warning' | 'failed' | 'crashed';
+  meta?: Record<string, unknown>;
+}
+
+interface HealthCheckResponse {
+  finishedAt: string;
+  checkResults: HealthCheckResult[];
+}
+
+export default class HealthController {
+  public async check(req: Request, res: Response): Promise<void> {
+    // Validate Oh Dear health check secret
+    const providedSecret = req.header('oh-dear-health-check-secret');
+    const expectedSecret = process.env['OH_DEAR_HEALTH_CHECK_SECRET'];
+
+    if (!expectedSecret) {
+      throw new ExternalServiceError('OH_DEAR_HEALTH_CHECK_SECRET environment variable is not set');
+    }
+
+    if (providedSecret !== expectedSecret) {
+      res.status(403).json({ error: 'Invalid health check secret' });
```

Comment:
```suggestion
      res.status(401).json({ error: 'Invalid health check secret' });
```
403 wäre: das secret stimmt aber der Client darf das nicht (sehen). Solange die credentials falsch sind sollte es glaube ich 401 sein

Reply:
da war ich mir die Nacht auch etwas unsicher.